src/java/main/org/apache/zookeeper/server/quorum/LearnerHandler.java

```diff
@@ -758,6 +760,11 @@ public boolean syncFollower(long peerLastZxid, ZKDatabase db, Leader leader) {
                 currentZxid = maxCommittedLog;
                 needOpPacket = false;
                 needSnap = false;
+            } else if (peerLastEpoch != lastProcessedEpoch && !db.isInCommittedLog(peerLastZxid)) {
```

Comment:
Could you please add a description to the comments above (to "Here are the cases that we want to handle") what this else if case is doing?